modules/@apostrophecms/i18n/index.js

```diff
@@ -65,6 +65,11 @@ module.exports = {
         }
         req.locale = locale;
         req.mode = mode;
+        if ((req.mode === 'draft') && (!self.apos.permission.can(req, 'view-draft'))) {
```

Comment:
Can this be part of permissions or `doc` middleware?

Reply:
Permissions should be checked immediately at the point where the danger begins, and this is one of the two places the mode can potentially be set. I do not want to move this to another handler/middleware function where a possibility exists that something will be inserted between.